### PR TITLE
fix(text-field): Fix accessbility issue with helper text

### PR DESCRIPTION
**Pull request by williamernest**
fixes: #2583

### CODE REVIEW

File: packages/mdc-textfield/helper-text/foundation.js [hunk 1]
```diff
@@ -87,37 +102,21 @@ class MDCTextFieldHelperTextFoundation extends MDCFoundation {
     }
   }
 
-  /** Makes the helper text visible to the screen reader. */
-  showToScreenReader() {
```
Comment (patrickrodee) [inline packages/mdc-textfield/helper-text/foundation.js]: Remove this from the README if it's no longer included.

File: packages/mdc-textfield/helper-text/foundation.js [hunk 2]
```diff
@@ -58,6 +59,20 @@ class MDCTextFieldHelperTextFoundation extends MDCFoundation {
     super(Object.assign(MDCTextFieldHelperTextFoundation.defaultAdapter, adapter));
   }
 
+  /**
+   * @return {boolean} True if helper text is a validation message.
+   */
+  isValidationMessage() {
```
Comment (patrickrodee) [inline packages/mdc-textfield/helper-text/foundation.js]: Add to README

File: packages/mdc-textfield/helper-text/foundation.js [hunk 3]
```diff
@@ -58,6 +59,20 @@ class MDCTextFieldHelperTextFoundation extends MDCFoundation {
...
+    return this.adapter_.hasClass(cssClasses.HELPER_TEXT_VALIDATION_MSG);
+  }
+
+  /**
+   * @return {string} id value of the helper text element
+   */
+  getId() {
```
Comment (patrickrodee) [inline packages/mdc-textfield/helper-text/foundation.js]: Add to README

File: packages/mdc-textfield/helper-text/foundation.js [hunk 4]
```diff
@@ -48,6 +48,7 @@ class MDCTextFieldHelperTextFoundation extends MDCFoundation {
       setAttr: () => {},
       removeAttr: () => {},
       setContent: () => {},
+      getId: () => {},
```
Comment (patrickrodee) [inline packages/mdc-textfield/helper-text/foundation.js]: Add to README

File: packages/mdc-textfield/foundation.js [hunk 5]
```diff
@@ -365,11 +368,24 @@ class MDCTextFieldFoundation extends MDCFoundation {
...
+      this.getNativeInput_().removeAttribute('aria-invalid');
     } else {
       this.adapter_.addClass(INVALID);
+      this.getNativeInput_().setAttribute('aria-invalid', true);
     }
+
     if (this.helperText_) {
       this.helperText_.setValidity(isValid);
+
+      if (isValid) {
```
Comment (patrickrodee) [inline packages/mdc-textfield/foundation.js]: The nesting here can be reduced to improve legibility. I'd collapse the conditionals.

```
if (isValid && !this.initiallyHadDescribedBy_) {
} else if (!this.initiallyHadDescribedBy_) {}
```

**Comment by williamernest** [comment]
Closing because this was already addressed by @rlfriedman in late May.

**Pull request closed without merging**